libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -442,6 +461,9 @@ export class ShellOperationRunner implements IOperationRunner {
                 projectChangeAnalyzer: this._projectChangeAnalyzer,
                 phaseName: this._phase.name
               });
+              this._projectBuildCache?.addAdditionalOutputFilePaths(
```

Comment:
We'll be best served by adding an entire folder to all operations that can be used for rush to dump stuff, e.g. the build log, the new state file, etc.

Reply:
That would be great! But i think it involves more design and maybe in another MR?

Because now if Rush sees `phase.isSynthetic`, the logout is `xxx.build.log`(not under an individual folder ❌), otherwise the logouts are under `rush-log`, which is also not a good name to cover this new state file.

On the other hand, the state file in this MR is new. It can be well designed. Now I chose to save the file under `.rush/temp/operation/<logIdentifier>` folder. I am open to relocate/rename the folder to be used for Rush to dump all the stuff.